examples/network/git2.c

```diff
@@ -57,16 +49,42 @@ int main(int argc, char **argv)
 
 	git_libgit2_init();
 
+	struct args_info args = ARGS_INFO_INIT;
+	const char *git_dir = NULL;
+
+	for (args.pos = 1; args.pos < args.argc; ++args.pos) {
+		char *a = args.argv[args.pos];
+
+		if (a[0] != '-') {
+			/* non-arg */
+			break;
+		} else if (optional_str_arg(&git_dir, &args, "--git-dir", ".git")) {
+		} else if (!strcmp(a, "--")) {
+			/* arg separator */
+			break;
+		}
+	}
+
+	// Before running the actual command, create an instance of the local
+	// repository and pass it to the function.
+
+	error = git_repository_open(&repo, git_dir);
+	if (error < 0)
+		repo = NULL;
+
 	for (i = 0; commands[i].name != NULL; ++i) {
-		if (!strcmp(argv[1], commands[i].name)) {
-			return_code = run_command(commands[i].fn, --argc, ++argv);
+		if (!strcmp(args.argv[args.pos], commands[i].name)) {
+			return_code = run_command(commands[i].fn, repo, args);
 			goto shutdown;
 		}
 	}
 
 	fprintf(stderr, "Command not found: %s\n", argv[1]);
 
 shutdown:
+	if(repo)
```

Comment:
This check is not required. `git_repository_free` already returns if `repo == NULL`.